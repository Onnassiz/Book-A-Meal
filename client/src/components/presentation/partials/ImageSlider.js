import React, { Component } from 'react';
import ImageGallery from 'react-image-gallery';
import 'react-image-gallery/styles/css/image-gallery.css';

import image1 from '../../../../assets/images/homeSlider/image_1.jpg';
import image2 from '../../../../assets/images/homeSlider/image_2.jpg';
import image3 from '../../../../assets/images/homeSlider/image_3.jpg';
import image4 from '../../../../assets/images/homeSlider/image_4.jpg';

class ImageSlider extends Component {
	constructor(props) {
		super(props);
		this.state = {
			images: [],
		};
	}

	componentWillMount() {
		const images = [
			{
				original: image3,
			},
			{
				original: image2,
			},
			{
				original: image1,
			},
			{
				original: image4,
			},
		];
		this.setState({ images });
	}

	render() {
		return (
			<div id="imageSlider">
				<ImageGallery
					items={this.state.images}
					slideInterval={6000}
					showThumbnails={false}
					autoPlay
					showPlayButton={false}
				/>
			</div>
		);
	}
}

export default ImageSlider;
