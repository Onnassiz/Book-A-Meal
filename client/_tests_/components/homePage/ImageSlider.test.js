import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImageSlider from '../../../src/components/presentation/partials/ImageSlider';

const wrapper = shallow(<ImageSlider />);

describe('<ImageSlider />', () => {
  it('Render User Home Page Image Slider component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
