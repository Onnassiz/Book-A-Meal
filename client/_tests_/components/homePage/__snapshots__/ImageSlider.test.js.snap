// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSlider /> Render User Home Page Image Slider component 1`] = `
<div
  id="imageSlider"
>
  <ImageGallery
    autoPlay={true}
    disableArrowKeys={false}
    disableSwipe={false}
    disableThumbnailScroll={false}
    flickThreshold={0.4}
    indexSeparator=" / "
    infinite={true}
    items={
      Array [
        Object {
          "description": "We ensure that better food is accessible to everyone.",
          "original": "https://res.cloudinary.com/onnassiz/image/upload/v1535468243/image_1_owy23d.jpg",
        },
        Object {
          "description": "We bring warm catering service to your doorstep.",
          "original": "https://res.cloudinary.com/onnassiz/image/upload/v1535468242/image_2_dmtjye.jpg",
        },
        Object {
          "description": "Let us worry about your diet.",
          "original": "https://res.cloudinary.com/onnassiz/image/upload/v1535468242/image_3_qiux12.jpg",
        },
        Object {
          "description": "Our caterers are willing to work on your recipe.",
          "original": "https://res.cloudinary.com/onnassiz/image/upload/v1535468243/image_4_ogto3u.jpg",
        },
        Object {
          "description": "Many caterers to give you diverse experiences.",
          "original": "https://res.cloudinary.com/onnassiz/image/upload/v1535468244/image_5_ln3fgi.jpg",
        },
      ]
    }
    lazyLoad={false}
    preventDefaultTouchmoveEvent={false}
    renderFullscreenButton={[Function]}
    renderLeftNav={[Function]}
    renderPlayPauseButton={[Function]}
    renderRightNav={[Function]}
    showBullets={true}
    showFullscreenButton={true}
    showIndex={false}
    showNav={false}
    showPlayButton={true}
    showThumbnails={false}
    slideDuration={450}
    slideInterval={6000}
    startIndex={0}
    stopPropagation={false}
    swipeThreshold={30}
    swipingTransitionDuration={0}
    thumbnailPosition="bottom"
    useBrowserFullscreen={true}
    useTranslate3D={true}
  />
</div>
`;
